frontend/public/components/routes.tsx

```diff
@@ -200,26 +201,24 @@ const RouteListRow: React.SFC<RoutesRowProps> = ({obj: route}) => <ResourceRow o
   <div className="col-lg-2 hidden-md hidden-sm hidden-xs"><RouteStatus obj={route} /></div>
 </ResourceRow>;
 
-const TLSSettings = props => <span>
-  {!props.tls && 'TLS is not enabled.'}
-  {props.tls && <dl>
-    <dt>Termination Type</dt>
-    <dd>{props.tls.termination}</dd>
-    <dt>Insecure Traffic</dt>
-    <dd>{props.tls.insecureEdgeTerminationPolicy || '-'}</dd>
-    <dt>Certificate</dt>
-    <dd>{props.tls.certificate || '-'}</dd>
-    <dt>Key</dt>
-    <dd>{props.tls.key || '-'}</dd>
-    <dt>CA Certificate</dt>
-    <dd>{props.tls.caCertificate || '-'}</dd>
-    {_.get(props.tls, 'termination') === 'reencrypt' && <span>
-      <dt>Destination CA Cert</dt>
-      <dd>{props.tls.destinationCACertificate || '-'}</dd>
-    </span>}
-  </dl>
+export const TLSSettings: any = props => {
+
+  if (!props.tls) {
+    return 'TLS is not enabled.';
   }
-</span>;
+
+  let secrets = _.omit(props.tls, ['termination', 'insecureEdgeTerminationPolicy']);
+
+  return <React.Fragment>
+    <SecretData heading="TLS Settings" data={secrets} type="route.openshift.io"/>
```

Comment:
Switching to SecretData also changes properties to alphabetical order, which is not really what we want here.